### PR TITLE
Modify EditCommand to accept NRIC instead of Index

### LINKED ISSUE #56: Modify EditCommand to accept NRIC instead of Index
Users will be able to edit specific patient details via a unique identifier instead of their index

### PR DESCRIPTION
**Pull request by hweinian**
closes #56 

Modified EditCommand to use nric instead of index.

Changes to delete: @yong-zhuo deleteCommand MESSAGE_INVALID_PERSON_DISPLAYED_INDEX messages was change to MESSAGE_PERSON_NRIC_NOT_FOUND

Test cases are disabled for the time being. It will be fixed in the next PR.

Let me know if there are any changes to be made!

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -145,4 +148,14 @@ public boolean equals(Object other) {
...
     }
 
+    //=========== Filtered Person List Accessors =============================================================
+    @Override
+    public Optional<Person> fetchPersonIfPresent(NricMatchesPredicate nricPredicate) {
+        List<Person> allPersonList = addressBook.getPersonList();
+
+        return allPersonList.stream()
+                .filter(nricPredicate::test)
+                .findFirst();
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/model/ModelManager.java:158]: SLAP is adhered to properly

File: src/main/java/seedu/address/model/person/NricMatchesPredicate.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        NricMatchesPredicate predicate = (NricMatchesPredicate) other;
+        return nric.equals(predicate.nric);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).toString();
+    }
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/model/person/NricMatchesPredicate.java:45]: Do create test cases for these methods to ensure CodeCov passes

**Review by yong-zhuo — COMMENTED** [review]
Overall LGTM 👍🏻

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 3]
```diff
@@ -38,10 +39,10 @@ public EditCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_NRIC, PREFIX_DOB, PREFIX_GENDER,
                         PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
 
-        Index index;
+        Nric patientNric;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            patientNric = ParserUtil.parseNric(argMultimap.getPreamble());
```
Comment (PresleyChew) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:45]: Clear naming. Good job!

File: src/main/java/seedu/address/model/person/NricMatchesPredicate.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        NricMatchesPredicate predicate = (NricMatchesPredicate) other;
+        return nric.equals(predicate.nric);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).add("nric", nric).toString();
+    }
+}
```
Comment (PresleyChew) [inline src/main/java/seedu/address/model/person/NricMatchesPredicate.java:46]: NricMatchesPredicate class well written!

**Review by PresleyChew — COMMENTED** [review]
Overall LGTM, do ensure code coverage by including more test cases thanks for the amazing work !

File: src/test/java/seedu/address/model/person/NricMatchesPredicateTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Test
+    public void toStringMethod() {
+
+        Nric nric = new Nric("T1234567A");
+        NricMatchesPredicate predicate = new NricMatchesPredicate(nric);
+        String expected = NricMatchesPredicate.class.getCanonicalName() + "{nric=" + nric + "}";
+
+        assertEquals(expected, predicate.toString());
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/model/person/NricMatchesPredicateTest.java:60]: Test cases are comprehensive and follows the style of existing predicate test cases

File: src/test/java/seedu/address/model/person/NricMatchesPredicateTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class NricMatchesPredicateTest {
+
+    @Test
+    public void equals() {
+        Nric nric = new Nric("T1234567A");
+        Nric nric2 = new Nric("T1234567B");
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/model/person/NricMatchesPredicateTest.java:16]: I don't think we should be using numbers in our variable names? Could consider naming it something like `firstNric` and `secondNric`

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 7]
```diff
@@ -53,46 +53,46 @@ public class EditCommand extends Command {
...
             + PREFIX_EMAIL + "johndoe@example.com";
 
-    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Patient: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
-
-    private final Index index;
+    public static final String MESSAGE_DUPLICATE_PERSON = "This patient already exists in the address book.";
+    private static final Logger logger = LogsCenter.getLogger(EditCommand.class);
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:63]: Good use of logging

**Pull request merged**